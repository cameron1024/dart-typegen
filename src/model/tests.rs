use super::*;

macro_rules! parse_snapshot {
    ($name:ident) => {
        #[test]
        fn $name() {
            let text = include_str!(crate::test_file!($name));
            let config = Library::parse_impl(None, text).unwrap();

            insta::assert_debug_snapshot!(config);
        }
    };
}

crate::all_test_files!(parse_snapshot);
