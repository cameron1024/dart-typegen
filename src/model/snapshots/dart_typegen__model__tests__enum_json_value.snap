---
source: src/model/tests.rs
expression: ron
---
Library {
    preamble: None,
    postamble: None,
    defaults: None,
    enums: [
        Enum {
            span: Span(
                0,
                0,
            ),
            name: SpannedScalar {
                value: "Color",
                span: SourceSpan {
                    offset: SourceOffset(
                        43,
                    ),
                    length: 7,
                },
            },
            docs: None,
            variants: [
                EnumVariant {
                    name: SpannedScalar {
                        value: "red",
                        span: SourceSpan {
                            offset: SourceOffset(
                                63,
                            ),
                            length: 5,
                        },
                    },
                    docs: None,
                    json_value: Some(
                        Value {
                            type_name: None,
                            literal: Spanned {
                                span: Span(
                                    87,
                                    103,
                                ),
                                value: String(
                                    "something-else",
                                ),
                            },
                        },
                    ),
                },
                EnumVariant {
                    name: SpannedScalar {
                        value: "green",
                        span: SourceSpan {
                            offset: SourceOffset(
                                118,
                            ),
                            length: 7,
                        },
                    },
                    docs: None,
                    json_value: Some(
                        Value {
                            type_name: None,
                            literal: Spanned {
                                span: Span(
                                    143,
                                    146,
                                ),
                                value: Int(
                                    Integer(
                                        Dec,
                                        "123",
                                    ),
                                ),
                            },
                        },
                    ),
                },
                EnumVariant {
                    name: SpannedScalar {
                        value: "blue",
                        span: SourceSpan {
                            offset: SourceOffset(
                                161,
                            ),
                            length: 6,
                        },
                    },
                    docs: None,
                    json_value: Some(
                        Value {
                            type_name: None,
                            literal: Spanned {
                                span: Span(
                                    185,
                                    189,
                                ),
                                value: Bool(
                                    true,
                                ),
                            },
                        },
                    ),
                },
            ],
        },
    ],
    classes: [],
    unions: [],
}
